proposals/csharp-12.0/collection-expressions.md

```diff
@@ -255,6 +255,52 @@ ImmutableArray<int> ia =
     List<int> z = [];
     ```
 
+## Spreads
+[spreads]: #spreads
+
+```csharp
+bool b = true;
+int[] x = [.. []];            // x = []
+byte[] y = [.. b ? [1] : []]; // y = [(byte)1]
+```
+
+A *spread element* `..E` is applicable as an iterator of type `T` if any of the following hold:
+
+* There is an implicit conversion from the [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) of `E` to `T`.
+* `E` is a *collection expression* and for each element `Eᵢ` there is an implicit conversion to `T`.
+* `E` is a [*conditional expression*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#1115-conditional-operator) `b ? x : y` where one or both of `x` and `y` is a *collection expression* and for each of the operands `x` and `y` one of the following hold:
```

Comment:
Would there be a way to obtain this behavior for conditional expressions without mentioning conditional expressions, so that it "just works" for switch expressions and other constructs? Maybe even covers nested collection expressions too?

For instance maybe piggy-backing off how target typing works through these expressions?

Reply:
That would be great :)

Reply:
Sorry, I missed reply from you in Teams:

> It doesn't seem unreasonable to me to explicitly list the set of constructs this works with. It'll be a short list. And for any new constructs in the future, we'll probably want to explicitly decide to support this or not.

Switch expressions are already potentially interesting to me.

What about generic method invocations? Could we avoid inferring `<int>` on `Iif` here?

```cs
long[] x = [42, ..Iif(cond, [43], [])];

ROS<T> Iif<T>(bool, ROS<T>, ROS<T>) => ...
```

Reply:
Note: i would not expect the IIf case to work.  Specifically, because i wouldn't think this would work either:

```
ROS<int> r = IIf(cond, [1, 2, 3], []);
```

We don't have type inference flow in from return types inwards.  As such, i wouldn't expect anything in the language would help with ths sort of case.  If, of course, we ended up supporting htat sort of outerwards-in version of type inference, then it would be nice if this just got picked up.